homeassistant/components/vallox/__init__.py

```diff
@@ -170,44 +190,33 @@ async def async_update_data() -> ValloxState:
             schema=service_details.schema,
         )
 
-    hass.data[DOMAIN] = {"client": client, "coordinator": coordinator, "name": name}
-
-    async def _async_load_platform_delayed(*_: Any) -> None:
-        # We need a successful update before loading the platforms, because platform init code
-        # derives the UUIDs from the data the coordinator fetches.
-        warned_once = False
-        while hass.state == CoreState.running:
-            await coordinator.async_refresh()
-            if coordinator.last_update_success:
-                break
-
-            if not warned_once:
-                _LOGGER.warning(
-                    "Vallox integration not ready yet; Retrying in background"
-                )
-                warned_once = True
-
-            await asyncio.sleep(INITIAL_COORDINATOR_UPDATE_RETRY_INTERVAL_SECONDS)
-        else:
-            return
+    await coordinator.async_config_entry_first_refresh()
 
-        hass.async_create_task(async_load_platform(hass, "sensor", DOMAIN, {}, config))
-        hass.async_create_task(async_load_platform(hass, "fan", DOMAIN, {}, config))
-
-    # The Vallox hardware expects quite strict timings for websocket requests. Timings that machines
-    # with less processing power, like a Raspberry Pi, cannot live up to during the busy start phase
-    # of Home Asssistant.
-    #
-    # Hence, wait for the started event before doing a first data refresh and loading the platforms,
-    # because it usually means the system is less busy after the event and can now meet the
-    # websocket timing requirements.
-    hass.bus.async_listen_once(
-        EVENT_HOMEASSISTANT_STARTED, _async_load_platform_delayed
-    )
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        "client": client,
+        "coordinator": coordinator,
+        "name": name,
+    }
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
 
     return True
 
 
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+        if len(hass.data[DOMAIN]) != 0:
```

Comment:
This could be simplified to just check for hass.data[DOMAIN]

Reply:
Like this?

```diff
diff --git homeassistant/components/vallox/__init__.py homeassistant/components/vallox/__init__.py
index f3379d7469..fe2d0d4d92 100644
--- homeassistant/components/vallox/__init__.py
+++ homeassistant/components/vallox/__init__.py
@@ -211,7 +211,7 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
         hass.data[DOMAIN].pop(entry.entry_id)

-        if len(hass.data[DOMAIN]) != 0:
+        if hass.data[DOMAIN]:
             return unload_ok

         for service in SERVICE_TO_METHOD:
```

Reply:
Reverse it with a not